### PR TITLE
Added Dedup Operator Documentation

### PR DESCRIPTION
**Pull request by bhupeshchawda**

### CODE REVIEW

File: docs/operators/dedup/dedup_operator_documentation.md [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+
+This document is intended as a guide for understanding and using the
+Dedup operator/module. It will guide the reader on the functionality and
+usage of Dedup operator / module.
+
+Terminology
+-----------
+
+We will refer to this operator as the Deduper or Dedup operator
+interchangeably.
```
Comment (chandnisingh) [inline docs/operators/dedup/dedup_operator_documentation.md]: Don't think we need another section for this. This statement can be added to the one above it.

**Comment by chandnisingh** [comment]
@amberarrow please review. 
The document looks good to me.

**Comment by bhupeshchawda** [comment]
Such a custom callback is not provided right now. The user will have to implement the processTuple() method to process the tuple, not just for the comparison.

**Comment by bhupeshchawda** [comment]
Addressed review comments.

File: docs/operators/dedup/dedup_operator_documentation.md [hunk 2]
```diff
@@ -0,0 +1,663 @@
+# Dedup - Operator Documentation
+
+
+Introduction
+============
+
+About this document
+-------------------
```
Comment (chandnisingh) [inline docs/operators/dedup/dedup_operator_documentation.md]: How about removing the 'Introduction' section and making 'About this document' the first heading.

There is just one sub-section under 'Introduction' which makes it look incomplete

**Pull request merged**